syncopy/tests/test_specest_fooof.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+#
+# Test FOOOF integration from user/frontend perspective.
+
+
+import pytest
+import numpy as np
+
+# Local imports
+from syncopy import freqanalysis
+from syncopy.shared.tools import get_defaults
+from syncopy.datatype import SpectralData
+from syncopy.shared.errors import SPYValueError
+from syncopy.tests.test_specest import _make_tf_signal
+
+import matplotlib.pyplot as plt
+
+
+def _plot_powerspec(freqs, powers):
+    """Simple, internal plotting function to plot x versus y."""
+    plt.plot(freqs, powers)
+    plt.xlabel('Frequency (Hz)')
+    plt.ylabel('Power (db)')
+    plt.show()
+
+
+class TestFooofSpy():
+    """
+    Test the frontend (user API) for running FOOOF. FOOOF is a post-processing of an FFT, and
+    to request the post-prcocesing, the user sets the method to "mtmfft", and the output to
+    one of the available FOOOF output types.
+    """
+
+    # Construct input signal
+    nChannels = 2
+    nChan2 = int(nChannels / 2)
+    nTrials = 1
+    seed = 151120
+    fadeIn = None
+    fadeOut = None
+    tfData, modulators, even, odd, fader = _make_tf_signal(nChannels, nTrials, seed,
```

Comment:
Is this really a good test signal for fooof? I can't se any 1/f characteristic for example.. also the frequency of the 'real' oscillatory component is not given anywhere! This pretty generic `_make_tf_signal` is almost deprecated (as shown again here, it's hard to see what it actually is), maybe try one of the guys residing in `tests/synth_data.py` instead ;)
EDIT: With `synth_data.AR2_network(AdjMat=np.zeros(1), alphas=[0.7, 0])` you get an AR(1) process which often serves as a good approximation of 1/f Now if you add some `synth_data.harmonic(...)` to it we can construct a suitable signal for testing.. Alternatively we'll do all that in the backend tests